src/main/controllers/SubscriptionUrnSearchResultController.ts

```diff
@@ -1,32 +1,28 @@
 import { Response } from 'express';
 import {PipRequest} from '../models/request/PipRequest';
-import {cloneDeep} from 'lodash';
 import {SubscriptionService} from '../service/subscriptionService';
+import validateRendering from '../common/utils';
 
 const subscriptionService = new SubscriptionService();
-
+let searchInput;
 export default class SubscriptionUrnSearchResultController {
 
   public async get(req: PipRequest, res: Response): Promise<void> {
-    const searchInput = req.query['search-input'];
+    searchInput = req.query['search-input'];
 
     if (searchInput && searchInput.length) {
       const searchResults = await subscriptionService.getSubscriptionUrnDetails(searchInput.toString());
 
-
-      if (searchResults) {
-        res.render('subscription-urn-search-results', {
-          ...cloneDeep(req.i18n.getDataByLanguage(req.lng)['subscription-urn-search-results']),
-          searchInput : searchInput,
-          searchResults: searchResults,
-        });
-      } else {
-        res.render('error', req.i18n.getDataByLanguage(req.lng).error);
-      }
+      validateRendering(searchResults,'subscription-urn-search-results',req, res, searchInput);
 
     }
     else {
       res.render('error', req.i18n.getDataByLanguage(req.lng).error);
     }
   }
+
+  public async post(req: PipRequest, res: Response): Promise<void> {
+    res.redirect(`subscription-confirmation?search-input=${searchInput}&stype=urn`);
```

Comment:
can this live in the forms post if this is all thats needed?

Reply:
this has been refactored

Reply:
refactored